openshift.py

```diff
@@ -0,0 +1,750 @@
+        return suggested_project_name
+
+    def get_request(self, url, debug=False):
+        r = requests.get(url, headers=self.headers, verify=self.verify)
+        if debug == True:
+            self.logger.info("url: " + url)
+            self.logger.info("g: " + str(r.status_code))
+            self.logger.info("g: " + r.text)
+        return r
+
+    def del_request(self, url, payload, debug=False):
+        if payload is None:
+            r = requests.delete(url, headers=self.headers, verify=self.verify)
+        else:
+            r = requests.delete(
+                url, headers=self.headers, data=json.dumps(payload), verify=self.verify
+            )
+        if debug == True:
+            self.logger.info("url: " + url)
+            if payload is not None:
+                self.logger.info("payload:" + json.dumps(payload))
+            self.logger.info("d: " + str(r.status_code))
+            self.logger.info("d: " + r.text)
+        return r
+
+    def put_request(self, url, payload, debug=False):
+        if payload is None:
+            r = requests.put(url, headers=self.headers, verify=self.verify)
+        else:
+            r = requests.put(
+                url, headers=self.headers, data=json.dumps(payload), verify=self.verify
+            )
+        if debug == True:
+            self.logger.info("url: " + url)
+            if payload is not None:
+                self.logger.info("payload:" + json.dumps(payload))
+            self.logger.info("pu: " + str(r.status_code))
+            self.logger.info("pu: " + r.text)
+        return r
+
+    def post_request(self, url, payload, debug=False):
+        r = requests.post(
+            url, headers=self.headers, data=json.dumps(payload), verify=False
+        )
+        if debug == True:
+            self.logger.info("url: " + url)
+            self.logger.info("payload: " + json.dumps(payload))
+            self.logger.info("po: " + str(r.status_code))
+            self.logger.info("po: " + r.text)
+        return r
+
+    def user_rolebinding_exists(self, user, project_name, role):
+        openshift_role = ""
+
+        if role == "admin":
+            openshift_role = "admin"
+        elif role == "member":
+            openshift_role = "edit"
+        elif role == "reader":
+            openshift_role = "view"
+        else:
+            return False
+
+        r = self.get_rolebindings(project_name, openshift_role)
+        if r.status_code == 200 or r.status_code == 201:
+            role_binding = r.json()
+            pprint.pprint(role_binding)
+            if (
+                "userNames" in role_binding.keys()
+                and role_binding["userNames"] is not None
+                and user in role_binding["userNames"]
+            ):
+                return True
+        return False
+
+    def get_all_moc_rolebindings(self, user, project_name):
+        rolebindings = []
+        for role in ["admin", "member", "view"]:
```

Comment:
You need to replace `view` with `reader` for this to work: the `user_rolebinding_exists` will return `False` if `role` isn't one of `['admin', 'member', 'reader']`.